src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -0,0 +1,94 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+import dataclasses
+from typing import Any, Final, TypeVar
+
+import numpy as np
+
+from functional.otf import languages, stages, step_types, workflow
+from functional.otf.source import cpp_gen, source
+from functional.program_processors.codegens.gtfn import gtfn_backend
+
+
+T = TypeVar("T")
+
+
+def get_param_description(name: str, obj: Any) -> source.ScalarParameter | source.BufferParameter:
+    view: np.ndarray = np.asarray(obj)
+    if view.ndim > 0:
+        return source.BufferParameter(name, tuple(dim.value for dim in obj.axes), view.dtype)
+    else:
+        return source.ScalarParameter(name, view.dtype)
+
+
+@dataclasses.dataclass(frozen=True)
+class GTFNSourceGenerator(
```

Comment:
I would like to see *translation* and/or *step* in the name due to the decision to call the ITIR->src step *translation*:
```suggestion
class GTFNTranslationStep(
```